ReactAndroid/src/main/java/com/facebook/react/packagerconnection/PackagerConnectionSettings.java

```diff
@@ -53,11 +53,24 @@ public String getDebugServerHost() {
     return host;
   }
 
-  public void setDebugServerHost(String host) {
-    mPreferences.edit().putString(PREFS_DEBUG_SERVER_HOST_KEY, host).apply();
+  public void setDebugServerHost(final String host) {
+    setDebugServerHost(host, true);
+  }
+
+  public void setDebugServerHost(final String host, final boolean savePrefs) {
```

Comment:
Minor suggestion, what about avoiding the boolean parameter by making this `setDebugServerHostWithoutPersisting` or something like that?

Reply:
the suggestion looks clearer to me. i've updated accordingly. thanks for the great suggestion and reviewing my prs.